### PR TITLE
feat: Added custom config to ds-card-reader

### LINKED ISSUE #25: Driver config -> app custom
ds-card-reader
ds-controller-board

### PR DESCRIPTION
**Pull request by marcpfuller**
Added custom configuration that closes #25

### CODE REVIEW

File: ds-card-reader/res/configuration.toml [hunk 1]
```diff
@@ -49,12 +49,12 @@ Type = "consul"
...
   Port = 59881
 
-[Driver]
-  DeviceName = 'card-reader'
-  DeviceSearchPath = '/dev/input/event*'
-  VID = '65535' # 0xFFFF
-  PID = '53'    # 0x0035
-  SimulateDevice = 'true'
+[DriverConfig]
+  DeviceName = "card-reader"
```
Comment (brian-intel) [inline ds-card-reader/res/configuration.toml:53]:
```suggestion
  DeviceName = "ds-card-reader"
```
ds-card-reader is the name expected by the other services.

Comment (brian-intel) [reply ds-card-reader/res/configuration.toml:53]: The /res/devices/device-list.toml "Name = 'ds-card-reader' also needs to be updated

**Review by brian-intel — CHANGES_REQUESTED** [review]

**Review by jim-wang-intel — APPROVED** [review]

File: ds-card-reader/device/config.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// Copyright © 2020 Intel Corporation. All rights reserved.
```
Comment (vli11) [inline ds-card-reader/device/config.go:1]: update copyright year

File: ds-card-reader/device/config_test.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
+// Copyright © 2020 Intel Corporation. All rights reserved.
```
Comment (vli11) [inline ds-card-reader/device/config_test.go:1]: update copyright year

**Review by vli11 — CHANGES_REQUESTED** [review]

**Review by brian-intel — APPROVED** [review]

File: ds-card-reader/go.mod [hunk 4]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/edgexfoundry/device-sdk-go/v2 v2.2.0
 	github.com/edgexfoundry/go-mod-core-contracts/v2 v2.2.0
 	github.com/gvalkov/golang-evdev v0.0.0-20180516222720-b6f418b1fe5a
-	github.com/intel-iot-devkit/automated-checkout-utilities v1.0.0
+	github.com/pkg/errors v0.9.1
```
Comment (lenny-intel) [inline ds-card-reader/go.mod:12]: So the use of this package changes the BOM for this RI. We have avoid it in KP as it isn't supported well and other issues. IMO we shouldn't use it here either.

File: ds-card-reader/res/devices/device-list.toml [hunk 5]
```diff
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
 [[DeviceList]]
-  Name = 'card-reader'
+  Name = 'ds-card-reader'
```
Comment (lenny-intel) [inline ds-card-reader/res/devices/device-list.toml:5]: The device name should not be the same as the service name. Please revert this chnage.

File: ds-card-reader/res/configuration.toml [hunk 6]
```diff
@@ -49,12 +49,12 @@ Type = "consul"
...
   Port = 59881
 
-[Driver]
-  DeviceName = 'card-reader'
-  DeviceSearchPath = '/dev/input/event*'
-  VID = '65535' # 0xFFFF
-  PID = '53'    # 0x0035
-  SimulateDevice = 'true'
+[DriverConfig]
+  DeviceName = "ds-card-reader"
```
Comment (lenny-intel) [inline ds-card-reader/res/configuration.toml:53]:
```suggestion
  DeviceName = "card-reader"
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: ds-card-reader/go.mod (hunk 4 above)
Comment (marcpfuller) [reply ds-card-reader/go.mod:12]: should I remove github.com/pkg/errors v0.9.1 and put github.com/intel-iot-devkit/automated-checkout-utilities v1.0.0 back?

Comment (lenny-intel) [reply ds-card-reader/go.mod:12]: No, errors is pulled in due to the use of errors.Wrap(). Avoid using this function.

Comment (marcpfuller) [reply ds-card-reader/go.mod]: changed all errors.Wrap() to fmt.errorf ()

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**